### PR TITLE
Add interacting-with-sap guide

### LINKED ISSUE #392: Create a guide for the SAP connector
**Description:**
Create a guide for demonstrating interactions with the SAP connector and add it to `docs/content/src/learn/guides/saas-clients` directory.

### PR DESCRIPTION
**Pull request by IrushiL**
resolves #392

### CODE REVIEW

File: docs/content/src/learn/tutorials/saas-integrations/sap/interacting-with-sap/README.md [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+- Ballerina Integrator
+- A Text Editor or an IDE
+    > **Tip**: For a better development experience, install the `Ballerina Integrator` extension in [VSCode](https://code.visualstudio.com/).
+- SAP R/3 instance with `sapidoc3.jar`, `sapjco3.jar` and the native SAP JCo library.
+
+## Implementation
+
+> If you want to skip the basics, you can download the git repo and directly move to the `Testing` section by skipping the `Implementation` section.
+
+1. Create a new project.
```
Comment (pramodya1994) [inline docs/content/src/learn/tutorials/saas-integrations/sap/interacting-with-sap/README.md]: Check these working properly in the site.

File: docs/content/src/learn/tutorials/saas-integrations/sap/interacting-with-sap/README.md [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+    password: config:getAsString("PASSWORD"),
+    ashost: config:getAsString("ASHOST"),
+    sysnr: config:getAsString("SYSNR"),
+    language: config:getAsString("LANGUAGE")
+};
+
+sap:Producer sapProducer = new (producerConfigs);
+```
+
+8. Functions `sendBapi()` and `sendIdoc()` could be used to send the BAPI requests and IDoc messages respectively.
```
Comment (samgnaniah) [inline docs/content/src/learn/tutorials/saas-integrations/sap/interacting-with-sap/README.md]:
```suggestion
8. Functions `sendBapi()` and `sendIdoc()` can be used to send the BAPI requests and IDoc messages respectively.
```

File: docs/content/src/learn/tutorials/saas-integrations/sap/interacting-with-sap/README.md [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+DESTINATION_NAME = "CPT"
+SAP_CLIENT = "800"
+USERNAME = "wso2_user"
+PASSWORD = "wso2_pass"
+ASHOST = "/H/217.116.29.154/S/3299/H/10.100.5.120/S/3200"
+SYSNR = "01"
+LANGUAGE = "en"
+```
+
+7. A SAP client could be configured as follows. Ballerina Integrator VS Code plugin contains a snippet for the same, which could be loaded using the autocomplete feature for `client/sap` keyword.
```
Comment (samgnaniah) [inline docs/content/src/learn/tutorials/saas-integrations/sap/interacting-with-sap/README.md]:
```suggestion
7. A SAP client can be configured as follows. Ballerina Integrator VS Code plugin contains a snippet for the same, which could be loaded using the autocomplete feature for `client/sap` keyword.
```

File: docs/content/src/learn/tutorials/saas-integrations/sap/interacting-with-sap/README.md [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+
+## Testing
+
+Let’s build the module. Navigate to the project directory and execute the following command.
+
+```bash
+ballerina build sap_producer
+```
+
+The build command would create an executable .jar file. Now run the .jar file created in the above step. Path to the `ballerina.conf` could be provided using the `--b7a.config.file` option.
```
Comment (samgnaniah) [inline docs/content/src/learn/tutorials/saas-integrations/sap/interacting-with-sap/README.md]:
```suggestion
The build command creates an executable .jar file. Now run the .jar file created in the above step. Path to the `ballerina.conf` could be provided using the `--b7a.config.file` option.
```

**Pull request merged**